lib/puppet/type/cs_colocation.rb

```diff
@@ -73,20 +74,53 @@ def should=(value)
   end
 
   autorequire(:cs_shadow) do
-    [ @parameters[:cib] ]
+    autos = []
+    if @parameters[:cib]
+      autos << @parameters[:cib].value
+    end
+
+    autos
+  end
+
+  if Puppet::PUPPETVERSION.to_f >= 4.0
```

Comment:
Have we ever discussed this antipattern in any venue? If not, I should take that to the dev list "soon". It's making me do a double take whenever I see it.